packages/grid/src/GridMetricCalculator.ts

```diff
@@ -319,47 +325,54 @@ export class GridMetricCalculator {
     );
 
     // Calculate some metrics for the scroll bars
-    const hasHorizontalBar = lastLeft > 0;
+    const hasHorizontalBar = lastLeft > 0 || contentWidth > viewportWidth;
     const hasVerticalBar = lastTop > 0;
     const horizontalBarHeight = hasHorizontalBar ? scrollBarSize : 0;
     const verticalBarWidth = hasVerticalBar ? scrollBarSize : 0;
     const barWidth = width - rowHeaderWidth - verticalBarWidth;
     const barHeight = height - columnHeaderHeight - horizontalBarHeight;
 
-    const handleWidth =
-      lastLeft > 0
-        ? Math.min(
-            Math.max(
-              minScrollHandleSize,
-              barWidth * ((columnCount - lastLeft) / columnCount)
-            ),
-            barWidth - 1
-          )
-        : 0;
-    const handleHeight =
-      lastTop > 0
-        ? Math.min(
-            Math.max(
-              minScrollHandleSize,
-              barHeight * ((rowCount - lastTop) / rowCount)
-            ),
-            barHeight - 1
-          )
-        : 0;
+    // How big the horizontal is relative to the horizontal bar
```

Comment:
We should be doing this for both horizontal and vertical scroll bars. Row height can be resized to be larger than the viewport as well (though not a use case we currently have).

Reply:
We are doing this for vertical bar currently, I just didn't rename it like I did horizontal. Do you want me to rename vertical too so it's more clear?